[Docs] Fix warnings in `mkdocs build` (continued)

Zerohertz opened the pull request:
Continued #23649

---

> [!NOTE]
> Warnings fixed:

```bash
WARNING -  griffe: vllm/core/block/naive_block.py:210: Failed to get 'name: description' pair from 'in whole allocator.'
WARNING -  griffe: vllm/core/block/prefix_caching_block.py:64: Parameter 'block_ids(Optional[Iterable[int]],' does not appear in the function signature

WARNING -  griffe: vllm/core/scheduler.py:660: Failed to get 'name: description' pair from 'that are currently running'

WARNING -  griffe: vllm/v1/attention/backends/flash_attn.py:441: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

WARNING -  griffe: vllm/v1/attention/backends/flashinfer.py:641: Parameter '#' does not appear in the function signature
WARNING -  griffe: vllm/v1/attention/backends/flashinfer.py:642: Parameter '#' does not appear in the function signature

WARNING -  griffe: vllm/v1/worker/utils.py:280: Failed to get 'name: description' pair from 'layers with layer names as keys.'
WARNING -  griffe: vllm/v1/worker/utils.py:174: Confusing indentation for continuation line 8 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/worker/utils.py:176: Confusing indentation for continuation line 10 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/worker/utils.py:177: Confusing indentation for continuation line 11 in docstring, should be 4 * 2 = 8 spaces, not 6

WARNING -  griffe: vllm/v1/core/kv_cache_manager.py:61: No type or annotation for returned value 1
WARNING -  griffe: vllm/v1/core/kv_cache_manager.py:62: No type or annotation for returned value 2
WARNING -  griffe: vllm/v1/core/kv_cache_manager.py:63: No type or annotation for returned value 3

WARNING -  griffe: vllm/v1/worker/tpu_model_runner.py:567: Failed to get 'name: description' pair from 'to be scheduled for each request.'
WARNING -  griffe: vllm/v1/worker/tpu_model_runner.py:570: No type or annotation for returned value 1
WARNING -  griffe: vllm/v1/worker/tpu_model_runner.py:571: No type or annotation for returned value 'contains'

WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:258: Confusing indentation for continuation line 6 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:260: Confusing indentation for continuation line 8 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:306: Confusing indentation for continuation line 11 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:308: Confusing indentation for continuation line 13 in docstring, should be 4 * 2 = 8 spaces, not 6

WARNING -  griffe: vllm/v1/attention/backends/cpu_attn.py:494: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size * num_kv_heads * head_size]'

WARNING -  griffe: vllm/v1/attention/backends/flex_attention.py:692: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

WARNING -  griffe: vllm/v1/attention/backends/pallas.py:238: Failed to get 'name: description' pair from 'kv_cache = [num_blocks, block_size, num_kv_heads * 2, head_size]'
WARNING -  griffe: vllm/v1/attention/backends/pallas.py:332: Failed to get 'name: description' pair from 'kv_cache = [num_blocks, block_size, num_kv_heads * 2, head_size]'

WARNING -  griffe: vllm/v1/attention/backends/rocm_aiter_fa.py:432: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

WARNING -  griffe: vllm/v1/attention/backends/tree_attn.py:365: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

WARNING -  griffe: vllm/v1/attention/backends/triton_attn.py:288: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

WARNING -  griffe: vllm/v1/attention/backends/xformers.py:333: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

WARNING -  griffe: vllm/v1/core/kv_cache_coordinator.py:121: No type or annotation for parameter 'num_tokens'
WARNING -  griffe: vllm/v1/core/kv_cache_coordinator.py:121: Parameter 'num_tokens' does not appear in the function signature

WARNING -  griffe: vllm/v1/executor/ray_distributed_executor.py:72: No type or annotation for parameter 'scheduler_output'

WARNING -  griffe: vllm/v1/metrics/prometheus.py:44: No type or annotation for returned value 'Registry'

WARNING -  griffe: vllm/v1/sample/logits_processor/interface.py:83: Failed to get 'name: description' pair from 'batch_update is non-None iff there have been'
WARNING -  griffe: vllm/v1/sample/logits_processor/interface.py:84: Failed to get 'name: description' pair from 'changes to the batch makeup.'

WARNING -  griffe: vllm/v1/sample/rejection_sampler.py:70: Parameter 'bonus_token_ids_tensor' does not appear in the function signature

WARNING -  griffe: vllm/v1/sample/tpu/sampler.py:91: No type or annotation for parameter 'logits'
WARNING -  griffe: vllm/v1/sample/tpu/sampler.py:91: Parameter 'logits' does not appear in the function signature

WARNING -  griffe: vllm/v1/structured_output/backend_types.py:126: Confusing indentation for continuation line 5 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/structured_output/backend_types.py:112: Confusing indentation for continuation line 5 in docstring, should be 4 * 2 = 8 spaces, not 6

WARNING -  griffe: vllm/v1/worker/gpu_input_batch.py:529: No type or annotation for parameter 'empty_req_indices'
WARNING -  griffe: vllm/v1/worker/gpu_input_batch.py:529: Parameter 'empty_req_indices' does not appear in the function signature

WARNING -  griffe: vllm/v1/worker/gpu_model_runner.py:2955: Failed to get 'name: description' pair from 'correct size but uninitialized shape.'

WARNING -  griffe: vllm/v1/worker/worker_base.py:39: Failed to get 'name: description' pair from 'responsibilities'
```

Zerohertz committed:
docs: docstring in naive_block

WARNING -  griffe: vllm/core/block/naive_block.py:210: Failed to get 'name: description' pair from 'in whole allocator.'
WARNING -  griffe: vllm/core/block/prefix_caching_block.py:64: Parameter 'block_ids(Optional[Iterable[int]],' does not appear in the function signature

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in scheduler

WARNING -  griffe: vllm/core/scheduler.py:660: Failed to get 'name: description' pair from 'that are currently running'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in flash_attn & flashinfer

WARNING -  griffe: vllm/v1/attention/backends/flash_attn.py:441: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'
WARNING -  griffe: vllm/v1/attention/backends/flashinfer.py:641: Parameter '#' does not appear in the function signature
WARNING -  griffe: vllm/v1/attention/backends/flashinfer.py:642: Parameter '#' does not appear in the function signature

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in v1 worker utils

WARNING -  griffe: vllm/v1/worker/utils.py:280: Failed to get 'name: description' pair from 'layers with layer names as keys.'
WARNING -  griffe: vllm/v1/worker/utils.py:174: Confusing indentation for continuation line 8 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/worker/utils.py:176: Confusing indentation for continuation line 10 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/worker/utils.py:177: Confusing indentation for continuation line 11 in docstring, should be 4 * 2 = 8 spaces, not 6

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring and type hin in kv_cache_manager

WARNING -  griffe: vllm/v1/core/kv_cache_manager.py:61: No type or annotation for returned value 1
WARNING -  griffe: vllm/v1/core/kv_cache_manager.py:62: No type or annotation for returned value 2
WARNING -  griffe: vllm/v1/core/kv_cache_manager.py:63: No type or annotation for returned value 3

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring and type hint in tpu_model_runner

WARNING -  griffe: vllm/v1/worker/tpu_model_runner.py:567: Failed to get 'name: description' pair from 'to be scheduled for each request.'
WARNING -  griffe: vllm/v1/worker/tpu_model_runner.py:570: No type or annotation for returned value 1
WARNING -  griffe: vllm/v1/worker/tpu_model_runner.py:571: No type or annotation for returned value 'contains'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: indent in encoder_cache_manager

WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:258: Confusing indentation for continuation line 6 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:260: Confusing indentation for continuation line 8 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:306: Confusing indentation for continuation line 11 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/core/encoder_cache_manager.py:308: Confusing indentation for continuation line 13 in docstring, should be 4 * 2 = 8 spaces, not 6

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz commented on vllm/v1/attention/backends/flash_attn.py:
```diff
@@ -438,7 +438,8 @@ def forward(
             key: shape = [num_tokens, num_kv_heads, head_size]
             value: shape = [num_tokens, num_kv_heads, head_size]
-            kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]
+            kv_cache: KV cache tensor with shape 
+                [2, num_blocks, block_size, num_kv_heads, head_size]
```
@hmellor 
Which one is better?

```suggestion
            kv_cache: shape = [2, num_blocks, block_size, num_kv_heads, head_size]
```

hmellor replied:
Probably `kv_cache: shape = [2, num_blocks, block_size, num_kv_heads, head_size]` to match the others. We already know it's KV cache because the variable is named well and we already know it's a tensor because of tye type hint.

Zerohertz committed:
docs: docstring in cpu_attn

WARNING -  griffe: vllm/v1/attention/backends/cpu_attn.py:494: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size * num_kv_heads * head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in flex_attention

WARNING -  griffe: vllm/v1/attention/backends/flex_attention.py:692: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring format

Signed-off-by: Hyogeun Oh (오효근) <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in pallas

WARNING -  griffe: vllm/v1/attention/backends/pallas.py:238: Failed to get 'name: description' pair from 'kv_cache = [num_blocks, block_size, num_kv_heads * 2, head_size]'
WARNING -  griffe: vllm/v1/attention/backends/pallas.py:332: Failed to get 'name: description' pair from 'kv_cache = [num_blocks, block_size, num_kv_heads * 2, head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in rocm_aiter_fa

WARNING -  griffe: vllm/v1/attention/backends/rocm_aiter_fa.py:432: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in tree_attn

WARNING -  griffe: vllm/v1/attention/backends/tree_attn.py:365: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in triton_attn

WARNING -  griffe: vllm/v1/attention/backends/triton_attn.py:288: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in xformers

WARNING -  griffe: vllm/v1/attention/backends/xformers.py:333: Failed to get 'name: description' pair from 'kv_cache = [2, num_blocks, block_size, num_kv_heads, head_size]'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in kv_cache_coordinator

WARNING -  griffe: vllm/v1/core/kv_cache_coordinator.py:121: No type or annotation for parameter 'num_tokens'
WARNING -  griffe: vllm/v1/core/kv_cache_coordinator.py:121: Parameter 'num_tokens' does not appear in the function signature

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: type hint of ray_distributed_executor

WARNING -  griffe: vllm/v1/executor/ray_distributed_executor.py:72: No type or annotation for parameter 'scheduler_output'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: type hint of prometheus

WARNING -  griffe: vllm/v1/metrics/prometheus.py:44: No type or annotation for returned value 'Registry'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in logits_processor interface

WARNING -  griffe: vllm/v1/sample/logits_processor/interface.py:83: Failed to get 'name: description' pair from 'batch_update is non-None iff there have been'
WARNING -  griffe: vllm/v1/sample/logits_processor/interface.py:84: Failed to get 'name: description' pair from 'changes to the batch makeup.'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in rejection_sampler

WARNING -  griffe: vllm/v1/sample/rejection_sampler.py:70: Parameter 'bonus_token_ids_tensor' does not appear in the function signature

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in tpu sampler

WARNING -  griffe: vllm/v1/sample/tpu/sampler.py:91: No type or annotation for parameter 'logits'
WARNING -  griffe: vllm/v1/sample/tpu/sampler.py:91: Parameter 'logits' does not appear in the function signature

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: indent in structured_output backend_types

WARNING -  griffe: vllm/v1/structured_output/backend_types.py:126: Confusing indentation for continuation line 5 in docstring, should be 4 * 2 = 8 spaces, not 6
WARNING -  griffe: vllm/v1/structured_output/backend_types.py:112: Confusing indentation for continuation line 5 in docstring, should be 4 * 2 = 8 spaces, not 6

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: docstring in worker gpu_input_batch

WARNING -  griffe: vllm/v1/worker/gpu_input_batch.py:529: No type or annotation for parameter 'empty_req_indices'
WARNING -  griffe: vllm/v1/worker/gpu_input_batch.py:529: Parameter 'empty_req_indices' does not appear in the function signature

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: indent in worker gpu_model_runner

WARNING -  griffe: vllm/v1/worker/gpu_model_runner.py:2955: Failed to get 'name: description' pair from 'correct size but uninitialized shape.'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: indent in worker_base

WARNING -  griffe: vllm/v1/worker/worker_base.py:39: Failed to get 'name: description' pair from 'responsibilities'

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
docs: indent in logits_processor interface

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
style: ruff

Signed-off-by: Zerohertz <ohg3417@gmail.com>

Zerohertz committed:
Merge branch 'main' into docs/mkdocs-warnings

hmellor reviewed (approved):
LGTM thanks for continuing this effort!
